src/network.cpp

```diff
@@ -266,141 +379,173 @@ namespace xiloader
+                std::cin >> globals::g_Username;
+                std::cout << "Password (6-32 characters): ";
+                globals::g_Password.clear();
+                std::cin >> globals::g_Password;
                 std::cout << "Repeat Password           : ";
                 std::cin >> input;
                 std::cout << std::endl;
 
-                if (input != g_Password)
+                if (input != globals::g_Password)
                 {
                     xiloader::console::output(xiloader::color::error, "Passwords did not match! Please try again.");
                     goto create_account;
                 }
 
-                sendBuffer[0x20] = 0x20;
+                sendBuffer[0x39] = 0x20;
             }
 
             std::cout << std::endl;
         }
         else
         {
             /* User has auto-login enabled.. */
-            sendBuffer[0x20] = 0x10;
+            sendBuffer[0x39] = 0x10;
             bFirstLogin = false;
         }
 
-        /* Copy username and password into buffer.. */
-        memcpy(sendBuffer + 0x00, g_Username.c_str(), 16);
-        memcpy(sendBuffer + 0x10, g_Password.c_str(), 16);
+        sendBuffer[0] = 0xFF; // Magic for new xiloader bits
 
-        /* Send info to server and obtain response.. */
-        send(sock->s, sendBuffer, 33, 0);
-        recv(sock->s, recvBuffer, 16, 0);
+        sendBuffer[1] = 0x00; // Feature flags, none used yet.
+        sendBuffer[2] = 0x00;
+        sendBuffer[3] = 0x00;
+        sendBuffer[4] = 0x00;
+        sendBuffer[5] = 0x00;
+        sendBuffer[6] = 0x00;
+        sendBuffer[7] = 0x00;
+        sendBuffer[8] = 0x00;
 
-        /* Handle the obtained result.. */
-        switch (recvBuffer[0])
-        {
-        case 0x0001: // Success (Login)
-            xiloader::console::output(xiloader::color::success, "Successfully logged in as %s!", g_Username.c_str());
-            sock->AccountId = *(UINT32*)(recvBuffer + 0x01);
-            closesocket(sock->s);
-            sock->s = INVALID_SOCKET;
-            return true;
+        /* Copy username and password into buffer.. */
+        memcpy(sendBuffer + 0x09, globals::g_Username.c_str(), globals::g_Username.length());
+        memcpy(sendBuffer + 0x19, globals::g_Password.c_str(), globals::g_Password.length());
 
-        case 0x0002: // Error (Login)
-            xiloader::console::output(xiloader::color::error, "Failed to login. Invalid username or password.");
-            closesocket(sock->s);
-            sock->s = INVALID_SOCKET;
-            return false;
+        /* Copy changed password into buffer */
+        memcpy(sendBuffer + 0x40, new_password.c_str(), 16);
 
-        case 0x0003: // Success (Create Account)
-            xiloader::console::output(xiloader::color::success, "Account successfully created!");
-            closesocket(sock->s);
-            sock->s = INVALID_SOCKET;
-            return false;
+        // 17 byte wide reserved space starting at 0x50
```

Comment:
I saw you talking about this in the other PR, is this for feature flags, or was this to allow servers that want to harvest _that bit of machine data that isn't that useful_ to slot it in here? If thats the case, I'd mark this as "operator-specific data" etc., rather than just reserved.

If this isn't for that then ignore me, carry on

Reply:
Done, you were correct.